lib/phase/flatten.js

```diff
@@ -228,6 +228,26 @@ function mergeTemplateIntoJob(jobName, jobConfig, newJobs, templateFactory) {
                 SD_TEMPLATE_VERSION: template.version
             });
 
+            // merge shared steps into oldJob
+            if (sharedConfig.steps
+                && (oldJob.annotations['screwdriver.cd/mergeStep']
```

Comment:
I think `mergeSharedSteps` is easier for users to understand. It may not be necessary considering that the only steps to merge are the values that defined in `shared`, but the user needs to fully understand how it works to use them, whereas with `mergeSharedSteps`, the user doesn't have to think about it at all.

Reply:
Change annotations to `mergeSharedSteps`.